cmd/segment-reaper/observer.go

```diff
@@ -143,11 +168,87 @@ func (obsvr *observer) processSegment(ctx context.Context, path metainfo.ScopedP
 // analyzeProject analyzes the objects in obsv.objects field for detecting bad
 // segments and writing them to objs.writer.
 func (obsvr *observer) analyzeProject(ctx context.Context) error {
-	// TODO this part will be implemented in next PR
-	// TODO(if): For what is this?
+	for bucket, objects := range obsvr.objects {
+		for path, object := range objects {
+			if object.skip {
+				continue
+			}
+
+			brokenObject := false
+			if object.hasLastSegment {
+				if object.expectedNumberOfSegments == 0 {
+					if !object.segments.IsSequence() {
+						brokenObject = true
+					}
+				} else if object.segments.Count() != int(object.expectedNumberOfSegments)-1 {
+					// expectedNumberOfSegments-1 because 'segments' doesn't contain last segment
+					brokenObject = true
+				}
```

Comment:
What if the number of segments matches, but they are not in a good sequence, i.e. the bitmask is something like `111010010011`? Should we check `object.segments.InSequence` too?

Reply:
I wasn't sure about using `InSequence` here but I think it needs to be added. Also I think we need to cover here case you were mentioning earlier that we have segements from older version of file. If overal number of segments found are bigger then `object.expectedNumberOfSegments` this might be case where we need to verify if only first segments are in sequence and number of them are equal to expected value.